x-pack/plugin/esql/compute/test/src/main/java/org/elasticsearch/compute/test/AnyOperatorTestCase.java

```diff
@@ -105,37 +114,107 @@ public final void testSimpleToString() {
-                } else if (operator instanceof SinkOperator) {
-                    assertThat(map, hasKey("pages_received"));
-                    assertThat(map, hasKey("rows_received"));
-                } else {
-                    assertThat(map, either(hasKey("pages_processed")).or(both(hasKey("pages_received")).and(hasKey("pages_emitted"))));
-                    assertThat(map, hasKey("rows_received"));
-                    assertThat(map, hasKey("rows_emitted"));
-                }
+            assertOperatorStatus(operator, List.of(), List.of());
+        }
+    }
+
+    /**
+     * Assert the operator has a status, and its values are acceptable.
+     * Delegates specific checks to {link #checkOperatorStatusFields}, which may be overridden.
+     */
+    protected void assertOperatorStatus(Operator operator, List<Page> input, List<Page> output) {
+        Operator.Status status = operator.status();
+
+        if (status == null) {
+            // Operator doesn't provide a status
+            return;
+        }
+
+        var xContent = XContentType.JSON.xContent();
+        try (var xContentBuilder = XContentBuilder.builder(xContent)) {
+            status.toXContent(xContentBuilder, ToXContent.EMPTY_PARAMS);
+
+            var bytesReference = BytesReference.bytes(xContentBuilder);
+            var map = XContentHelper.convertToMap(bytesReference, false, xContentBuilder.contentType()).v2();
+
+            // For some operators, we don't know here if they consumed/processed all the input.
+            // That check should be done in the operator test
+            var nonNegativeMatcher = input.isEmpty() ? matchNumberEqualTo(0) : matchNumberGreaterThanOrEqualTo(0);
+            var inputPagesMatcher = matchNumberEqualTo(input.size());
+            var totalInputRows = input.stream().mapToLong(Page::getPositionCount).sum();
+            var inputRowsMatcher = matchNumberEqualTo(totalInputRows);
+            var outputPagesMatcher = matchNumberEqualTo(output.size());
+            var totalOutputRows = output.stream().mapToLong(Page::getPositionCount).sum();
+            var outputRowsMatcher = matchNumberEqualTo(totalOutputRows);
+
+            if (operator instanceof SourceOperator) {
+                assertThat(map, hasEntry(is("pages_emitted"), outputPagesMatcher));
+                assertThat(map, hasEntry(is("rows_emitted"), outputRowsMatcher));
+            } else if (operator instanceof SinkOperator) {
+                assertThat(map, hasEntry(is("pages_received"), inputPagesMatcher));
+                assertThat(map, hasEntry(is("rows_received"), inputRowsMatcher));
+            } else if (operator instanceof AsyncOperator) {
+                assertThat(map, hasEntry(is("received_pages"), nonNegativeMatcher));
+                assertThat(map, hasEntry(is("completed_pages"), nonNegativeMatcher));
+                assertThat(map, hasEntry(is("process_nanos"), nonNegativeMatcher));
```

Comment:
The async operator status fields are serialized as "received_pages" instead of "pages_received" (Like the others operators). Should I change it here? It's not even a transport version change, just some unit tests